dashboard/app/models/sections/section.rb

```diff
@@ -282,6 +282,34 @@ def toggle_hidden_script(script, should_hide)
     end
   end
 
+  # One of the contstraints for teachers looking for discount codes is that they
+  # have a section in which 10+ students have made progress on 5+ levels in both
+  # csd2 and csd3
+  def has_sufficient_discount_code_progress?
+    return false if students.length < 10
+    csd2 = Script.get_from_cache('csd2')
+    csd3 = Script.get_from_cache('csd3')
+    return false unless csd2 && csd3
```

Comment:
I wonder if this should raise an error, rather than just returning false - it seems like something is seriously wrong if we're querying for discount code progress and the csd2 and 3 scripts don't exist.

Reply:
I suspect these scripts will not exist in all of our test environments, which is why I added this check. Maybe we still do want to throw in those cases though, since this isn't code that will be called unless we're testing this feature, at which point we will want these scripts.